src/app/components/topnavigation/topnavigation.component.ts

```diff
@@ -9,13 +9,13 @@ import { Component, OnInit, Input } from '@angular/core';
 export class TopnavigationComponent implements OnInit {
 
   @Input() bookstore_name: string;
-  @Input() cart: any;
-  @Input() hamburger: any;
+  @Input() cart: string;
+  @Input() hamburger: string;
```

Comment:
Input are usually used to send data between different components: for example parent and child components. however, here you use for the same component.
Instead of @Input, you can just declare them to variables such as: public bookstore_name = 'Bookstore' inside class and call them in templates/views.